python/tests/unit/arcticdb/version_store/test_normalization.py

```diff
@@ -478,6 +478,59 @@ def test_no_inplace_index_array_modification(lmdb_version_store, sym, datetime64
     assert pandas_container.index.dtype == datetime64_dtype
 
 
+def test_index_names_roundtrip_datetime(lmdb_version_store, sym):
+    df = pd.DataFrame(data={"a": [1, 2, 3]}, index=pd.DatetimeIndex(["2020-01-01", "2020-01-02", "2020-01-03"]))
+    new_index = pd.Timestamp("2020-01-01")
+    df.index.rename(new_index, inplace=True)
+    lmdb_version_store.write(sym, df)
+    df.index.rename(str(new_index), inplace=True)
+    assert_frame_equal(lmdb_version_store.read(sym).data, df)
+    assert df.index.names[0] == str(new_index)
```

Comment:
I thought we would just validate against these ones that we can't get to roundtrip properly, just with a better error message than the current:

```
InternalException: pybind11::cast_error(Unable to cast Python instance of type <class 'list'> to C++ type 'std::vector<std::string, std::allocator<std::string> >')
```

I think it will get quite hard to work with ArcticDB unless we keep the invariant that,

```
lib.write("s", df)
pandas.testing.assert_frame_equal(lib.read("s").data, df)
```

Reply:
We could do it like that, but for the column names, we already implicitly parse and use them as strings (unless they are ints).
If we add this verification, it could result in breaking code that worked before.